### ISSUE TITLE
ArrayBracketNoWhitespace - right square bracket before right curly. Can this be optional? Is it a false positive?

### ISSUE THREAD

**Issue by MateuszStefek** [checkstyle/checkstyle#21345]
I have read check documentation: https://checkstyle.sourceforge.io/checks/whitespace/arraybracketnowhitespace.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash

/var/tmp $ cat config.xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE module PUBLIC "-//Puppy Crawl//DTD Check Configuration 1.3//EN" "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">

	<module name="TreeWalker">
		<module name="ArrayBracketNoWhitespace" />
	</module>

</module>

/var/tmp $ cat MyClass.java
public class MyClass {
	public static void main(String[] args) {
		String[] oddParams = { args[1], args[3], args[5] };
	}
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-14.0.0-all.jar -c config.xml MyClass.java 
Starting audit...
[ERROR] [...]/MyClass.java:3:64: ']' is followed by whitespace. [ArrayBracketNoWhitespace]
Audit done.
Checkstyle ends with 1 errors.
```

---

ArrayBracketNoWhitespace complains about the space after `[5]`, before `}`.

This doesn't match the description of the check.

However, there's an example `  int[] warnArr = {arr[0], arr[1] };// violation ''\]' is followed by whitespace'` showing the exact same thing, so it is probably intentional.

I don't see how this behaviour was derived from Google Java Style guide. The style guide even permits something like this:

```quote
Optional just inside both braces of an array initializer
new int[] {5, 6} and new int[] { 5, 6 } are both valid
```

---
ArrayBracketNoWhitespace has found a lot of false positives in my project. Only this example is something I feel is not natural. IntelliJ's default auto-formatting agrees here.

If `String[] oddParams = { args[1], args[3], "x" };` is valid, then `String[] oddParams = { args[1], args[3], args[5] };` should also be valid, in my opinion.

If it is not a false positive, can we at least change this aspect of the check to be configurable?

**Comment by Atharv3221** [comment]
@romani 
```java
atharv@atharv-IdeaPad-3-15IIL05:~/test$ cat MyClass.java
/** Some javadoc. */
public class MyClass {

  /** Some javadoc. */
  public static void main(String[] args) {
    String[] oddParams = { args[1], args[3], args[5] };
  }
}
```

```
atharv@atharv-IdeaPad-3-15IIL05:~/test$ java -jar checkstyle-14.0.0-all.jar -c /google_checks.xml MyClass.java
Starting audit...
[WARN] /home/atharv/test/MyClass.java:6:52: ']' is followed by whitespace. [ArrayBracketNoWhitespace]
Audit done.
```

This is false positive for

[s4.6.2-horizontal-whitespace](https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace)
as
> 9. Optional just inside both braces of an array initializer
new int[] {5, 6} and new int[] { 5, 6 } are both valid

**Comment by M-SaaD-H** [comment]
Hi @MateuszStefek @Atharv3221,

According to the check requirements, the whitespace after `]` in `{ args[1], args[3], args[5] }` is not expected, so this is not a false positive.

The example mentioned from the Google Java Style Guide is:

```text
new int[] {5, 6} and new int[] { 5, 6 } are both valid
```

Both forms are valid because the `]` is outside the `{ }`, unlike the case currently being reported by the check.

This behavior was also discussed previously with Roman, and he confirmed that the whitespace after `]` is not expected in this case.

That said, I'm open to changing the behavior if we decide that the check requirements should be interpreted differently.
